### PR TITLE
RR-230 - Improvements to ITs for create/retrieve ActionPlan

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**
This PR adds some additional checks and changes to the create/retrieve Action Plan integration tests, which I realised were missing.

### CODE REVIEW

File: src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GetActionPlanTest.kt [hunk 1]
```diff
@@ -122,6 +135,8 @@ class GetActionPlanTest : IntegrationTestBase() {
     val actual = response.responseBody.blockFirst()
     assertThat(actual)
       .isForPrisonNumber(prisonNumber)
+      .hasReviewDateCategory(ReviewDateCategory.SIX_MONTHS)
+      .hasNoReviewDate()
```
Comment (mattwills-moj-digital) [inline src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GetActionPlanTest.kt:139]: The controversial bit! I'm not sure if I should automatically populate `reviewDate` if the `reviewDateCategory` is set to 3, 6 or 12 months 🤔 I've chosen not to as there are no requirements around this, but I'll try and find out more from the BAs.

Comment (mattwills-moj-digital) [reply src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GetActionPlanTest.kt:139]: mmm...the more I think about it, the more I think I should set the `reviewDate`, as I can't see any downsides to this approach. We'll still know which option the user selected (i.e. to populate the UI, or for reporting purposes).

In contrast, I can foresee problems with not setting it! 😂

Comment (nathanrussell-moj-digital) [reply src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GetActionPlanTest.kt:139]: Hmmm, I'm not so sure
My thinking is that you currently have it correct, and your test is perfect as its our safety net for and and when we might get asked to change the behaviour.

If the business tell us to calculate the `reviewDate` from the value of the enum, then thats fine. But I dont think we should 2nd guess that as a requirement. I think we should only do it if its documented / communicated as a business requirement and expected behaviour.

File: src/testFixtures/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/resource/model/ActionPlanResponseAssert.kt [hunk 2]
```diff
@@ -21,6 +22,36 @@ class ActionPlanResponseAssert(actual: ActionPlanResponse?) :
...
+
+  fun hasNoReviewDate(): ActionPlanResponseAssert {
+    isNotNull
+    with(actual!!) {
+      if (reviewDate != null) {
+        failWithMessage("Expected reviewDate to be null, but was $reviewDate")
+      }
+    }
+    return this
+  }
```
Comment (nathanrussell-moj-digital) [inline src/testFixtures/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/resource/model/ActionPlanResponseAssert.kt:53]: ❤️ 
I absolutely love our assertJ assertion classes 👍 ❤️

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM (as you currently have it, rather than deriving the `reviewDate`)

File: src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GetActionPlanTest.kt (hunk 1 above)
Comment (mattwills-moj-digital) [reply src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GetActionPlanTest.kt:139]: Let me talk you through this alternative approach: https://github.com/ministryofjustice/hmpps-education-and-work-plan-api/pull/66

**Pull request closed without merging**